crawler.py

```diff
@@ -540,49 +539,48 @@ def scrape_registrations(self, async=True):
         """
         self.debug_logger.info("Scraping registrations, async = " + str(async))
         if async:
-            self._scrape_pages(self.registration_urls)
+            self._scrape_pages('registration', self.registration_urls)
         else:
             for elem in self.registration_url_tuples:
                 lst = []
                 while len(self.registration_urls) > 0 and elem[0] in self.registration_urls:
                     lst.append(self.registration_urls.pop(0))
-                self._scrape_pages(lst)
+                self._scrape_pages('registration', lst)
         self.debug_logger.info("Finished scraping registrations, async = " + str(async))
 
     def scrape_users(self):
         """
         Wrapper method that scrape all urls in self.user_urls. Calls _scrape_pages().
         """
         self.debug_logger.info("Scraping users")
-        self._scrape_pages(self.user_urls)
+        self._scrape_pages('profile', self.user_urls)
         self.debug_logger.info("Finished scraping users")
 
     def scrape_institutions(self):
         """
         Wrapper method that scrape all institution_urls. Calls _scrape_pages().
         """
         self.debug_logger.info("Scraping institutions")
-        self._scrape_pages(self.institution_urls)
+        self._scrape_pages('institution', self.institution_urls)
         self.debug_logger.info("Finished scraping institutions")
 
     def scrape_general(self):
         """
         Wrapper method that scrape all general_urls. Calls _scrape_pages().
         """
         self.debug_logger.info("Scraping general pages")
-        self._scrape_pages(self.general_urls)
+        self._scrape_pages('', self.general_urls)
         self.debug_logger.info("Finished scraping general pages")
 
-    # TODO Make semaphore value a parameter
-    def _scrape_pages(self, aspect_list):
+    def _scrape_pages(self, osf_type, aspect_list, sem_value=5):
```

Comment:
It would be better if 'osf_type' is a optional parameter. So you don't have to pass a empty string to "scrape_general".